### ISSUE TITLE
sanitizeHtml throws TypeError on '&' symbol

### ISSUE THREAD

**Issue by matejfalat** [apostrophecms/sanitize-html#606]
## To Reproduce

Step by step instructions to reproduce the behavior:

```
sanitizeHtml('<p>&</p>')
// or
sanitizeHtml('<p>&nbsp</p>')
```

## Expected behavior
Not to crash.

## Describe the bug
When the html text contains the ampersand symbol, the `sanitizeHtml()` is failing with:
```
Uncaught TypeError: Cannot read properties of undefined (reading '0')
    at Tokenizer.stateBeforeEntity (Tokenizer.js?6fbd:582:1)
    at Tokenizer.parse (Tokenizer.js?6fbd:818:1)
    at Tokenizer.write (Tokenizer.js?6fbd:158:1)
    at Parser.write (Parser.js?5804:459:1)
    at sanitizeHtml (index.js?5e22:578:1)
    at MaterialPreviewPage (MaterialPreviewPage.tsx?d2f2:41:55)
```

## Details

**Version of Node.js:**
v18.13.0

**Server Operating System:**
Windows 11, WSL2, and Docker

**Screenshots**

![error1](https://user-images.githubusercontent.com/81639615/215270160-c8ad26c6-ce2b-4c46-ad76-d8eadb860fb3.png)

![error2](https://user-images.githubusercontent.com/81639615/215270161-6fb0ce53-6ea3-4b1f-b34d-74bca15e1c05.png)

**Comment by BoDonkey** [comment]
Hi @matejfalat,
I tested this on a mac using the repo tests and could not reproduce the error. Is this occurring only in the browser? I wonder if this is an `htmlparser2` issue, rather than a `sanitize-html` issue since `Tokenizer` is part of that package. I guess we would need some minimal project set-up to replicate this error.
Cheers

**Comment by boutell** [comment]
Yes, what we would ask is that you contribute a PR with a failing unit test to this repo so we can see how this is possible in the context of this project and avoid any confusion with issues that might only exist in a larger project with parts that aren't actually dependencies of the module etc. htmlparser2 is a dependency so browser or no, a bug coming from that should be reproducible in a test.

**Comment by victorbojica** [comment]
probably because of missing ";" at the end

**Comment by NewEraCracker** [comment]
On Node.js CLI:
```
> const sanitizeHtml = require('sanitize-html');
undefined
> sanitizeHtml('<p>&</p>')
'<p>&amp;</p>'
> sanitizeHtml('<p>&nbsp</p>')
'<p> </p>'
>
```

Just tested this and it works good on both of these:
- With the patched version [2.7.3](https://github.com/apostrophecms/sanitize-html/issues/663#issuecomment-2249942595) (and htmlparser2 6.1.0);
- On another project using 2.13.0 (and htmlparser2 8.0.2).

Might be worth adding a test case. My two cents. Thank you.
